blocks/quote-carousel/quote-carousel.js

```diff
@@ -0,0 +1,162 @@
+    prevButton.removeAttribute('disabled');
+  } else if (nextIndex === 0) {
+    indexElement.dispatchEvent(event);
+    // disable previous button
+    prevButton.setAttribute('disabled', true);
+  } else if (nextIndex === (numChildren[key] - 1)) {
+    // disable next button
+    nextButton.setAttribute('disabled', true);
+    stopAutoScroll(block);
+  } else if (currentIndex === (numChildren[key] - 1)) {
+    // disable next button
+    nextButton.removeAttribute('disabled');
+  }
+  slidesContainer.children[currentIndex].removeAttribute('active');
+  slidesContainer.children[nextIndex].setAttribute('active', true);
+  slidesContainer.style.transform = `translateX(-${nextIndex * 100}%)`;
+}
+
+/**
+ * Start auto scroll
+ *
+ * @param {Element} block
+ * @param {number} interval
+ */
+function startAutoScroll(block, interval = DEFAULT_SCROLL_INTERVAL_MS) {
+  const key = getDataKey(block);
+  scrollInterval[key] = setInterval(() => {
+    const currentIndex = getCurrentSlideIndex(block);
+    switchSlide((currentIndex + 1) % numChildren[key], block);
+  }, interval);
+}
+
+/**
+ * Returns block content from the spreadsheet
+ *
+ * @param {string} type
+ * @returns {Promise<any>}
+ */
+async function getBlockContent(type) {
+  const url = `/drafts/quotes/helix-quotes.json?sheet=${type}`;
+  let data = { data: {} };
+  try {
+    const resp = await fetch(url);
+    data = await resp.json();
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error('Failed to load block content', error);
+  }
+  return data;
+}
+
+export default async function decorate(block) {
+  const key = getDataKey(block);
+  const content = await getBlockContent(key);
+  numChildren[key] = content.total;
+  block.querySelector('div:nth-child(1)').style.display = 'none';
+  // generate carousel content from loaded data
+  const slidesContainer = document.createElement('div');
+  slidesContainer.classList.add('carousel-content');
+
+  [...content.data].forEach((row) => {
+    const rowContent = document.createElement('div');
+    rowContent.classList.add('item');
+    rowContent.innerHTML = `
+                <p class="quote">"${row.quote}"</p>
+                <p class="author">${row.author}</p>
+                <p class="position">${row.position}</p>
+                `;
+    rowContent.classList.add('item');
+    slidesContainer.appendChild(rowContent);
+  });
+  slidesContainer.children[0].setAttribute('active', true);
+
+  // generate container for carousel controls
+  const controlsContainer = document.createElement('div');
+  controlsContainer.classList.add('controls-container');
+  controlsContainer.innerHTML = `
+    <div class="pagination">
+        <span class="index">1</span>
+        &nbsp;of&nbsp;
+        <span class="total">${numChildren[key]}</span>
+    </div>
+    <button name="prev" aria-label="Previous" class="control-button" disabled><svg><use xlink:href="../../icons/arrow.svg#carrot"/></svg></button>
```

Comment:
I'd suggest using a fully qualified path to the svg, rather than relative.

Reply:
updated

Reply:
Only one got updated?